unvis.go

```diff
@@ -4,12 +4,13 @@ package mtree
 import "C"
 import "fmt"
 
+// Unvis is a wrapper for the C implementation of unvis, which decodes a string
+// that potentially has characters that are encoded with Vis
 func Unvis(str string) (string, error) {
-	dst := new(C.char)
+	dst := C.CString(string(make([]byte, len(str)+1)))
 	ret := C.strunvis(dst, C.CString(str))
-	if ret == 0 {
+	if ret == -1 {
```

Comment:
similar story here. It's the length. So the returning of an `error` is not entirely needed.